devenv.sh

```diff
@@ -11,4 +11,27 @@ then
   eval $(minikube docker-env)
 else
   echo "WARNING: minikube is not started - cannot change docker-env"
+
+  echo "Generating empty kubeconfig file"
+  KUBECONFIG_FILE=./empty-kubeconfig
```

Comment:
Can you add it to .gitignore?

Reply:
Done (amended existing patch).

Reply:
@obnoxxx Oh I think my comment was too quick.

I have two changes to suggest:
1. Keep that empty-kubeconfig file in git under `build/`, and then devenv will just update the KUBECONFIG to point to it. The .gitignore change is uneeded for this.
2. I think we should prefer that devenv will not override existing KUBECONFIG unless there is really no configuration available, because it can interfere with working on remote clusters. Can we detect this case with `kubectl config view` - I am not sure what error it gives when the configuration is empty in your minikube stop case.

What do you think?

Reply:
@guymguym  - Sounds good. 

Question to clarify: Do you suggest to commit the empty kubeconfig file under `build/` or to generate there?

Reply:
I think to commit it makes more sense, since this is a static asset for building that we can also change if needed.

Reply:
Committing makes sense - will do that.

Regarding not overriding KUBECONFIG: checking for the variable is easy. But that might override a working minikube kube config, which is coming from `$HOME/.kube/config`.  But using `kubectl config view` does not seem to work:

The problem with `kubectl config view` is that it will spit out its config no matter whether the cluster is reachable or up. If minikube is stopped, it spits out the valid config. But the later commands triggered by `operator-sdk run` don't work if the cluster is not up.

Since we can build with the empty kubeconfig, the question would be: wouldn't it be fine to always build with this?

Reply:
Right. So we can pass this directly in the Makefile to the call `operator-sdk run --kubeconfig=build/empty-kubeconfig ...`

Reply:
I like it when things get more simple. :-)

Reply:
Updated. Only modified targets `cli` and `gen-olm`. `run` still requires a kubernetes env. The setting can still be overridden by the KUBCONFIG env variable. But building now works without.

Reply:
Magnificent.